### PR TITLE
Add /admin using exadmin

### PR DESCRIPTION
**Pull request by ericworkman**
@mbramson Pull this down and try it out. You'll need to modify a user to have is_admin = true, then navigate to /admin.

### CODE REVIEW

File: priv/repo/migrations/20170125185650_set_is_admin_default.exs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def up do
+    alter table(:users) do
+      modify :is_admin, :boolean, default: false
+    end
+
+    execute "UPDATE users SET is_admin = FALSE where is_admin IS NULL"
+  end
+
+  def down do
+  end
```
Comment (mbramson) [inline priv/repo/migrations/20170125185650_set_is_admin_default.exs:16]: We should probably define the down of not having a default for :is_admin.

File: priv/repo/migrations/20170125185650_set_is_admin_default.exs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+defmodule Pairmotron.Repo.Migrations.SetIsAdminDefault do
+  use Ecto.Migration
+
+  def up do
+    alter table(:users) do
+      modify :is_admin, :boolean, default: false
```
Comment (mbramson) [inline priv/repo/migrations/20170125185650_set_is_admin_default.exs]: Should we also make this field not-null? After the Update?

File: web/router.ex [hunk 3]
```diff
@@ -29,8 +34,14 @@ defmodule Pairmotron.Router do
     get "/", SessionController, :new
     post "/login", SessionController, :create
     get "/logout", SessionController, :delete
+    delete "/logout", SessionController, :delete # Exadmin logout needs this
   end
 
+	scope "/admin", ExAdmin do
+    pipe_through [:browser, :authenticate, :admin]
+    admin_routes
+	end
```
Comment (mbramson) [inline web/router.ex]: What's with the indentation here?

**Review by mbramson — COMMENTED** [review]
Noice. Some thoughts.

**Comment by mbramson** [comment]
Managed to get the following error by navigating to localhost:4000/admin and clicking on "user" on the left

```
Request: GET /admin/users
** (exit) an exception was raised:
    ** (ArgumentError) `toggle` option could be used only with columns of boolean type.
But `is_admin` is ["Data type": "Atom", "Reference modules": "Atom"]
with value == nil
        lib/ex_admin/helpers.ex:168: ExAdmin.Helpers.build_single_field/4
        lib/ex_admin/helpers.ex:134: ExAdmin.Helpers.build_field/4
        lib/ex_admin/themes/admin_lte2/index.ex:184: anonymous fn/4 in ExAdmin.Theme.AdminLte2.Index.build_table_body/4
        (elixir) lib/enum.ex:1623: Enum."-reduce/3-lists^foldl/2-0-"/3
        lib/ex_admin/themes/admin_lte2/index.ex:183: anonymous fn/5 in ExAdmin.Theme.AdminLte2.Index.build_table_body/4
        (elixir) lib/enum.ex:1184: Enum."-map/2-lists^map/1-0-"/2
        lib/ex_admin/themes/admin_lte2/index.ex:171: ExAdmin.Theme.AdminLte2.Index.build_table_body/4
        lib/ex_admin/themes/admin_lte2/index.ex:54: ExAdmin.Theme.AdminLte2.Index.paginated_collection_table/2
        lib/ex_admin/themes/admin_lte2/index.ex:152: ExAdmin.Theme.AdminLte2.Index.batch_action_form/6
        lib/ex_admin/themes/admin_lte2/index.ex:14: ExAdmin.Theme.AdminLte2.Index.wrap_index_grid/1
        (pairmotron) lib/ex_admin/index.ex:158: Pairmotron.ExAdmin.User.index_view/3
        web/controllers/admin_resource_controller.ex:25: ExAdmin.AdminResourceController.index/3
        web/controllers/admin_resource_controller.ex:1: ExAdmin.AdminResourceController.action/2
        web/controllers/admin_resource_controller.ex:1: ExAdmin.AdminResourceController.phoenix_controller_pipeline/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.instrument/4
        (pairmotron) lib/phoenix/router.ex:261: Pairmotron.Router.dispatch/2
        (pairmotron) web/router.ex:1: Pairmotron.Router.do_call/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.phoenix_pipeline/1
        (pairmotron) lib/plug/debugger.ex:123: Pairmotron.Endpoint."call (overridable 3)"/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.call/2
```

I'm also getting FontAwesome errors all over the place in console, although this isn't causing issues in the app because :erlang:.

```
Request: GET /fonts/fontawesome-webfont.ttf?v=4.6.2
** (exit) an exception was raised:
    ** (MatchError) no match of right hand side value: :error
        (pairmotron) web/controllers/page_controller.ex:19: Pairmotron.PageController.show/2
        (pairmotron) web/controllers/page_controller.ex:1: Pairmotron.PageController.action/2
        (pairmotron) web/controllers/page_controller.ex:1: Pairmotron.PageController.phoenix_controller_pipeline/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.instrument/4
        (pairmotron) lib/phoenix/router.ex:261: Pairmotron.Router.dispatch/2
        (pairmotron) web/router.ex:1: Pairmotron.Router.do_call/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.phoenix_pipeline/1
        (pairmotron) lib/plug/debugger.ex:123: Pairmotron.Endpoint."call (overridable 3)"/2
        (pairmotron) lib/pairmotron/endpoint.ex:1: Pairmotron.Endpoint.call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:15: Plug.Adapters.Cowboy.Handler.upgrade/4
        (cowboy) src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4
```

Otherwise this is awesome.

**Comment by ericworkman** [comment]
@mbramson made some of the improvements we discussed. Try running the improved migration

**Review by Serneum — DISMISSED** [review]
One file is named `authenticate` and the other is named `authorization`. D- for consistency

**Comment by mbramson** [comment]
I am still seeing the FontAwesome error when using this. Besides it appearing in the console, there don't seem to be any adverse effects, but I still think that we should figure it out before it becomes part of the codebase.

**Comment by ericworkman** [comment]
@mbramson Can you run `mix admin.install` and see if the font files are generated and the error goes away?

**Review by mbramson — APPROVED** [review]
Awesome. LGTM!

**Comment by ericworkman** [comment]
Thanks @Serneum @mbramson

**Pull request merged**